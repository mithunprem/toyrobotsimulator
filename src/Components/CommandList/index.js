import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const CommandList = ({ commandList }) => {
  return (
    <Fragment>
      {
        <div className="command-list">
          <div className="command-list-header pl-3 pt-2 pb-2">
            Commands executed
          </div>
          <table className="table m-0 table-sm table-striped table-bordered">
            <tbody>
              {
                commandList.map((command, index) => {
                  return(
                    <tr key={index}>
                      <td className="p-2 command-row">{command}</td>
                    </tr>
                  );
                })
              }
            </tbody>
          </table>
        </div>
      }
    </Fragment>
  );
}

CommandList.propTypes = {
  commandList: PropTypes.array
}

export default CommandList;
