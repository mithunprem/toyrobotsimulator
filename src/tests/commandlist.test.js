import React from 'react';
import { create } from 'react-test-renderer';
import CommandList from '../Components/CommandList';

describe("CommandList", () => {
  const commandListData = [ 'Place 0,0,north', 'move', 'report'];
  test("should render without throwing an error", () => {
    const commandList = create(<CommandList commandList={commandListData}/> ).toJSON()
    expect(commandList).toMatchSnapshot()
  });
});
