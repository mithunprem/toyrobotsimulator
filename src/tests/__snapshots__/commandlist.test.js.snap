// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandList should render without throwing an error 1`] = `
<div
  className="command-list"
>
  <div
    className="command-list-header pl-3 pt-2 pb-2"
  >
    Commands executed
  </div>
  <table
    className="table m-0 table-sm table-striped table-bordered"
  >
    <tbody>
      <tr>
        <td
          className="p-2 command-row"
        >
          Place 0,0,north
        </td>
      </tr>
      <tr>
        <td
          className="p-2 command-row"
        >
          move
        </td>
      </tr>
      <tr>
        <td
          className="p-2 command-row"
        >
          report
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
